src/utils.rs

```diff
@@ -0,0 +1,32 @@
+/* standard use */
+use std::io;
+
+/* project use */
+use crate::error::Error;
+use crate::seek::compression::ReadSeek;
+
+pub fn get_first_five<'a, T>(mut in_stream: T) -> Result<([u8; 5], T), Error>
+where
+    T: io::Read + 'a,
```

Comment:
Instead of requiring a concrete type you can just verify that the trait bounds are satisfied - then this no longer needs to be copied between the modules.